Changelog.md

```diff
@@ -16,6 +16,7 @@ Bugfixes:
  * SMTChecker: Fix internal error when using bitwise operators with an array element as argument.
  * Standard JSON Interface: Fix ICE when the optimizer is disabled and an empty/blank string is used for ``optimizerSteps`` sequence.
  * StaticAnalyzer: Only raise a compile time error for division and modulo by zero when it's between literals.
+ * TypeChecker: Fix segmentation fault when assigning to complex left side tuple expression that resolves to empty.
```

Comment:
This changelog entry will be hard to understand to users. I don't think it's clear that "resolving" here means removing useless parentheses. It's just our internal terminology used in the implementation, not official language term.

It's also incorrect because it's not about empty tuples after all.

I'd use this:
```suggestion
 * TypeChecker: Fix compiler crash when the left-hand side of an assignment was a parenthesized non-tuple expression of a tuple type.
```
I guess it's still hard to understand without knowing what I wrote in that long comment (https://github.com/ethereum/solidity/pull/15084#discussion_r1594499648), but at least should be correct :)